### ISSUE TITLE
Implicit upsampler with DINOv2 at non-square image size

### ISSUE THREAD

**Issue by ttk-kstn** [mhamilton723/FeatUp#34]
Hi, thank you for sharing the great work.

 The code looks like it can handle non-square images for DINOv2 input. However, I encountered an issue in `downsamplers.py` which I resolved by modifying the following code:
```python
        patches = torch.nn.Unfold(self.kernel_size, stride=stride)(inputs) \
            .reshape(
            (b, self.in_dim, self.kernel_size * self.kernel_size, self.final_size, self.final_size * int(w / h))) \
            .permute(0, 3, 4, 2, 1)
```

to:
```python
            (b, self.in_dim, self.kernel_size * self.kernel_size, self.final_size, int(self.final_size * (w / h)))) \
```

With this change, `train_implicit_sampler.py` worked for DINOv2 input sizes like 280x392.

Additionally, 392 seems to be the largest input size that this Unforld() & reshape work (at least for kernel_size=29). If there's ways to go beyond 392, please share.

Thanks,

**Comment by ttk-kstn** [comment]
After a bit of experimentation, I got FeatUp to work at DINOv2 input resolution of (406, 406).

My reasoning is that `Unfold()` in downsamplers should use the same stride as the patch embedding layers of ViT's. For example, [DINOv2 uses `Conv2D(stride=(14, 14))`](https://github.com/facebookresearch/dinov2/blob/e1277af2ba9496fbadf7aec6eba56e8d882d1e35/dinov2/layers/patch_embed.py#L65)
```python
self.proj = nn.Conv2d(in_chans, embed_dim, kernel_size=patch_HW, stride=patch_HW)
```

So I changed the down sampler code to:
```python
        stride = h // self.final_size  # 14 for DINOv2 
        padding = max(0, (self.kernel_size - stride + 1) // 2)
        patches = torch.nn.Unfold(self.kernel_size, padding=padding, stride=stride)(inputs) \
            .reshape(
            (b, self.in_dim, self.kernel_size * self.kernel_size, self.final_size, int(self.final_size * (w / h)))) \
            .permute(0, 3, 4, 2, 1)
```

The learned downsampler kernels looks file. (It was off-center when I got padding wrong at first).
![image](https://github.com/mhamilton723/FeatUp/assets/6270382/3f7294f3-37be-4132-a225-9c6aa1926025) 
![image](https://github.com/mhamilton723/FeatUp/assets/6270382/35c9d9ef-c236-4a1f-b646-9c4e503c1391)


I attached feat/hr/* I got with the above mod for one of the sample images.
![image](https://github.com/mhamilton723/FeatUp/assets/6270382/443a2fec-6b76-4937-a0f4-8755e502febd)
![image](https://github.com/mhamilton723/FeatUp/assets/6270382/bf1a3a0a-abae-4222-accd-d4c2ec6c58b4)
![image](https://github.com/mhamilton723/FeatUp/assets/6270382/5bef8c5a-29c3-4d89-b9c6-bc4fd8d6c034)

**Comment by Mr-CD1** [comment]
I want to know how to attach  feat/hr/* ,cause now I have the .pth file but don't know how to use it .Thanks
